src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -634,6 +589,28 @@ public JsonPropertyInfo CreateJsonPropertyInfo(Type propertyType, string name)
             return propertyInfo;
         }
 
+        [MemberNotNull(nameof(_properties))]
+        private void PopulatePropertyList()
```

Comment:
I think when this is called from `Configure` you can call `_isConfigued` path, currently `_isConfigured` changes after `Configure` is called. Consider taking `_isConfigured` as an argument rather than reading a field

Reply:
`Configure` is only accessing the underlying `_properties` field. It would be allocating the list unnecessarily if it did. I suppose I add a `!Monitor.IsEntered` assertion to make sure nobody does it accidentally.

Reply:
Done.